tests/generative/test_query_model.py

```diff
@@ -112,14 +156,46 @@ def run_test(query_engines, data, variable, recorder):
         ), f"Mismatch between {first_name} and {second_name}"
 
 
+def run_error_test(query_engines, data, variable):
+    instances, variables = setup_test(data, variable)
+    for name, engine in query_engines.items():
+        try:
+            run_with(engine, instances, variables)
+        except Exception as e:  # pragma: no cover
+            assert False, f"{name} engine encountered an error: {e}"
+
+
+IGNORED_ERRORS = [
+    # MSSQL can only accept 10 levels of CASE nesting. The variable strategy will sometimes
+    # generate queries that exceed that limit.
+    ".+Case expressions may only be nested to level 10.+",
+    # OUT-OF-RANGE DATES
+    # The variable strategy will sometimes result in date operations that contruct
+    # invalid dates (e.g. a large positive or negative integer in a DateAddYears operation
+    # may result in a date with a year that is outside of the allowed range)
+    # The different query engines report errors from out-of-range dates in different ways:
+    # mssql
+    ".+Cannot construct data type date.*",  # DateAddYears, with an invalid calculated year
+    ".+Adding a value to a 'date' column caused an overflow.+",  # DateAddMonths, resulting in an invalid date
+    # sqlite
+    "Couldn't parse date string",
+    # in-memory engine
+    "year -?\\d+ is out of range",  # DateAddYears, with an invalid calculated year
+    "date value out of range",  # DateAddMonths, resulting in an invalid date
+]
```

Comment:
I like the clear documentation of which error is which here.

It's a bit of a shame that the type of the exceptions is disconnected from the error messages. How about catching `Exception` and then testing the type and message against a list of expected pairs?